docs/pfcwd/PFCWD_BASIC.md

```diff
@@ -0,0 +1,100 @@
+
+Note that there is no way to clear PFC watchdog statistics unless we reload config database or minigraph. 
+
+The testbed consists of two IXIA ports and a SONiC device under test (DUT) as follows. All the IXIA ports should have the same bandwidth capacity. To reduce the configuration complexity, we recommond configuring the switch as a Top of Rack (ToR) / Tier 0 (T0) switch and binding three switch interfaces to the Vlan. PFC watchdog must be enabled at the DUT.
+
+```
+                        _________
+                       |         |
+IXIA port 1 ------ et1 |   DUT   | et2------ IXIA port 2
+                       |_________|
+
+```
+
+## Experiment Steps
+In this experiment, we need to create three traffic items:
+
+- Data traffic 1: Data packets from IXIA port 1 to IXIA port 2. The traffic demand is 100% line rate. The trafffic is mapped to the lossless priority i (e.g., 3 or 4) at the switch. To this end, packets should be marked with the correct DSCP value (e.g., DSCP 3 for priority 3).
+  
+- Data traffic 2: Data packets from IXIA port 1 to IXIA port 2. The traffic demand is 100% line rate. The trafffic is *also* mapped to the lossless priority i (e.g., 3 or 4) at the switch. To this end, packets should be marked with the correct DSCP value (e.g., DSCP 3 for priority 3). *Data traffic 2 and 1 have different time durations and start delays.* 
+
+- PFC pause storm: Persistent PFC pause frames from the IXIA port 2 to et2 of DUT having same priority (e.g., 3) as data traffic. The inter-frame transmission interval should be smaller than per-frame pause duration.
+
+This experiment needs the following steps. We need to repeat these steps for each lossless priority individually and all the lossless priorities simultaneously. Let’s use $T_{detect}$, $T_{restore}$, and $T_{poll}$ to denote the detection time, restoration time, and polling interval of PFC watchdog.  
+
+This experiment needs the following steps. We need to repeat these steps for each lossless priority individually and all the lossless priorities simultaneously. We should try two PFC storm durations: one is larger than $T_{detect} + T_{poll}$ (trigger PFC watchdog) and the other one is smaller than $T_{detect}$ (not trigger PFC watchdog)
+
+- At time 0, start PFC pause storm. PFC pause storm lasts for $T_{storm}$. 
+   
+- At time $T_{restore}/2$, start data traffic 1. The duration of data traffic 1 is also $T_{storm}$.
+
+- At time $T_{storm}$, stop PFC pause storm.
+  
+- At time $T_{restore}/2 + T_{storm}$, stop data traffic 1.
+  
+- At time $T_{restore} + T_{poll} + T_{storm}$, start data traffic 2. Its duration is 1 second. 
+ 
+-	At time $T_{restore} + T_{poll} + T_{storm}$ + 1 second, stop data traffic 2. 
+
+After the above steps, we should check the following items:
+
+- When $T_{storm}$ is larger than $T_{detect}$ + $T_{poll}$:
+  - PFC watchdog is triggered on the corresponding lossless priority of et2.
+  - All the packets of traffic 1 are dropped.
```

Comment:
Is this under the assumption that the PFC pause storm will entirely pause the queue. if not, some packets will still make it through till the storm is detected

Reply:
Yes. The inter-frame transmission interval of PFC storm should be smaller than per-frame pause duration (see description of PFC storm).

Reply:
> Is this under the assumption that the PFC pause storm will entirely pause the queue? if not, some packets will still make it through till the storm is detected

Yes. See description of PFC storm.

Reply:
The Traffic 1 is started at T_restore/2. I believe this is smaller than T_detect+T_poll, in which case some initial traffic will be passed through. Which will not match with the expectation that Traffic 1 will be 100% dropped.

Traffic 1 should start at (T_detect+T_poll) and stop at (T_storm - T_detect - T_poll + T_restore/2). This will make sure 100% traffic drop for Traffic 1.

Reply:
> The Traffic 1 is started at T_restore/2. I believe this is smaller than T_detect+T_poll, in which case some initial traffic will be passed through. Which does not match with the expectation that Traffic 1 should be 100% dropped.
> 
> Traffic 1 should start at (T_detect+T_poll) and run for the duration (T_storm - T_detect - T_poll + T_restore/2). This will make sure 100% traffic drop for Traffic 1.

This is wrong. Though traffic 1 starts before the triggering of PFC watchdog, it will be blocked by the PFC storm. Once PFC watchdog is triggered, all the packets queued in the switch will be dropped.